[TASK] Compatibility update to TYPO3 12 and 13

calien666 opened the pull request:
* Rewrite Services.php to Services.yaml
* remove deprecated renderStatic from ViewHelper
* add TYPO3 12.4 and 13.4 LTS compatible backend template registration
* drop TYPO3 11 support

calien666 committed:
[TASK] Make Ext core v12 compatible

commands:
```sh
composer req php:"^7.4 || ^8.0 || ^8.1 || ^8.2 || ^8.3" --no-update
composer req typo3/cms-core:"^11.5 || ^12.4" --no-update
composer req --dev \
  typo3/cms-extensionmanager:"^11.5 || ^12.4" \
  typo3/cms-fluid-styled-content:"^11.5 || ^12.4" \
  typo3/cms-fluid:"^11.5 || ^12.4" \
  typo3/cms-frontend:"^11.5 || ^12.4" \
  typo3/cms-info:"^11.5 || ^12.4" \
  typo3/cms-lowlevel:"^11.5 || ^12.4" \
  typo3/cms-tstemplate:"^11.5 || ^12.4" \
  typo3/coding-standards:^0.8 \
  -W --no-update

composer req --dev ssch/typo3-rector:^2.6 --no-update
```

calien666 committed:
[TASK] Update php and TYPO3 to 12 and 13

```
composer require php:"^8.1 || ^8.2 || ^8.3 || ^8.4" --no-update
composer require typo3/cms-core:"^12.4 || ^13.4" --no-update
composer require typo3/cms-backend:"^12.4 || ^13.4" --no-update
composer remove --dev typo3/cms-extensionmanager typo3/cms-fluid-styled-content typo3/cms-frontend typo3/coding-standards --no-update
composer remove --dev typo3/cms-fluid --no-update
composer require typo3/cms-fluid:"^12.4 || ^13.4" --no-update
composer require --dev typo3/cms-info:"^12.4 || ^13.4" --no-update
composer require --dev typo3/cms-lowlevel:"^12.4 || ^13.4" --no-update
composer require --dev typo3/cms-tstemplate:"^12.4 || ^13.4" --no-update
```

calien666 committed:
[TASK] Update runTests to latest version

Copied from fgtclb/typo3-file-required-attributes

calien666 committed:
[TASK] Remove outdated TypoScript configuration

With this pull request, the support for TYPO3 11 is dropped.

calien666 committed:
[TASK] Adjust github code testing actions

Remove old v11 testing and add v13 testing.
Streamline installation of correct versions.

calien666 committed:
[TASK] Streamline phpstan configuration

Add 13 phpstan files and remove useless checks

calien666 committed:
[TASK] Remove Core 11 phpstan configuration

calien666 committed:
[TASK] Adjust error handling

Check for empty request and change general runtimeException
with specific ones.

calien666 committed:
[TASK] Update php-cs-fixer configuration

calien666 committed:
[TASK] Run code style fixer

calien666 committed:
[TASK] Update README.md

Change the badges and refactor texts, as they were just wrong.

calien666 committed:
[TASK] Adjust documentation

* change registration information
* remove wrong documentation
* change rendering from Sphinx to PhpDoc

calien666 committed:
[TASK] Update ext_emconf.php

Change TYPO3 dependencies and increase version number

calien666 committed:
[TASK] Disable TypoScript linter

With changing from ext_typoscript to page.tsconfig registration,
the TypoScript linter is not needed for the moment.

sbuerk commented on ext_emconf.php:
```diff
@@ -7,11 +7,10 @@
         'depends' => [
-            'typo3' => '11.5.0-11.5.99',
+            'typo3' => '12.4.0-13.4.99',
         ],
```
backend + fluid should be added to be aligned with composer require

calien666 replied:
DOne.

sbuerk commented on composer.json:
```diff
@@ -24,21 +24,19 @@
-		"typo3/cms-lowlevel": "^11.5",
-		"typo3/cms-tstemplate": "^11.5",
-		"typo3/coding-standards": "^0.6.1"
+		"ssch/typo3-rector": "^2.6",
```
Why adding rector as permanent development dependency ?

Rector should only be used on point and not committed.

calien666 replied:
removed, your'e right.

sbuerk commented on rector.php:
```diff
@@ -0,0 +1,52 @@
+<?php
+
```
Rector should only be used on point and not committed. Like requireing it permenantly, committing the config should not be done and removed.

sbuerk commented on Documentation/Files/versionSupport.csv:
```diff
@@ -0,0 +1,2 @@
+"1.x","11",">=7.4"
```
Can't we have this version matrix already within the README.md file ?

sbuerk commented on .gitattributes:
```diff
@@ -15,6 +15,7 @@
 .phplint.yml export-ignore
 .stylelintrc export-ignore
 docker-compose.yaml export-ignore
+rector.php export-ignore
```
see below, rector should not be commited and based on that gitattributes exclude is not needed.

calien666 committed:
[BUGFIX] Adjust smaller issues

Since TYPO3 13 the registration of xml namespaces is no longer allowed
with `https`, so change to `http`.
Also adjust the registration path for backend view includes
